src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
+            if (nodeToUnlink.item == object || (nodeToUnlink.item != null
+                    && nodeToUnlink.item.equals(object))) {
+                unlink(nodeToUnlink);
+                return true;
+            }
+            nodeToUnlink = nodeToUnlink.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Can't find element with index " + index);
+        }
+    }
+
+    private Node<T> findByIndex(int index) {
+        int middleOfList = (size - 1) / 2;
+        if (index > middleOfList) {
+            return findElementInLastHalf(index);
+        } else {
+            return findElementInFirstHalf(index);
+        }
+    }
+
+    private Node<T> findElementInFirstHalf(int index) {
+        Node<T> nodeByIndex = first;
+        for (int i = 0; i < index; i++) {
+            nodeByIndex = nodeByIndex.next;
+        }
+        return nodeByIndex;
+    }
+
+    private Node<T> findElementInLastHalf(int index) {
+        Node<T> nodeByIndex = last;
+        for (int i = size - 1; i > index; i--) {
+            nodeByIndex = nodeByIndex.prev;
+        }
+        return nodeByIndex;
+    }
+
+    private void linkBefore(T value, Node<T> nodeByIndex) {
+        Node<T> prevNode = nodeByIndex.prev;
+        Node<T> newNode = new Node<>(prevNode, value, nodeByIndex);
+        nodeByIndex.prev = newNode;
+        if (prevNode == null) {
+            first = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkLast(T value) {
+        Node<T> lastNode = last;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        last = newNode;
+        if (lastNode == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
+    }
+
+    private T unlink(Node<T> nodeToUnlink) {
+        Node<T> nextNode = nodeToUnlink.next;
+        Node<T> prevNode = nodeToUnlink.prev;
+        if (prevNode == null) {
+            first = nextNode;
+        } else {
+            prevNode.next = nextNode;
+            nodeToUnlink.prev = null;
+        }
+        if (nextNode == null) {
+            last = prevNode;
+        } else {
+            nextNode.prev = prevNode;
+            nodeToUnlink.next = null;
+        }
+        T value = nodeToUnlink.item;
+        nodeToUnlink.item = null;
+        size--;
+        return value;
+    }
+
+    private class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
```

Comment:
I'm not sure, but I suppose it might be better to put this class after we fill the fields of our main class, but before methods. And it should be static, I think.